composable_pipeline_1.py

```diff
@@ -0,0 +1,109 @@
+"""Example of a composable pipeline"""
+
+import dlt
+import os
+import random
+from dlt.destinations import filesystem, duckdb
+
+# fixtures
+customers = [
+    {"id": 1, "name": "dave"},
+    {"id": 2, "name": "marcin"},
+    {"id": 3, "name": "anton"},
+    {"id": 4, "name": "alena"},
+]
+
+products = [
+    {"name": "apple", "price": 1},
+    {"name": "pear", "price": 2},
+    {"name": "banana", "price": 3},
+    {"name": "schnaps", "price": 10},
+]
+
+if __name__ == "__main__":
+    os.environ["DATA_WRITER__DISABLE_COMPRESSION"] = "True"
+
+    #
+    # 1. let's load some stuff to a duckdb pipeline (standin for a remote location)
+    #
+    duck_pipeline = dlt.pipeline(
+        pipeline_name="warehouse", destination=duckdb(credentials="warehouse.duckdb")
+    )
+
+    @dlt.resource(write_disposition="replace", table_name="customers")
+    def c():
+        yield from customers
+
+    @dlt.resource(write_disposition="replace", table_name="orders")
+    def o():
+        order_no = 0
+        # every customer orders 4 things everyday
+        for weekday in ["monday", "tuesday", "wednesday"]:
+            for customer in customers:
+                for i in range(4):
+                    order_no += 1
+                    product = random.choice(products)
+                    yield {
+                        "order_day": weekday,
+                        "id": order_no,
+                        "customer_id": customer["id"],
+                        "product": product["name"],
+                        "price": product["price"],
+                    }
+
+    # run and print result
+    print("RUNNING WAREHOUSE INGESTION")
+    print(duck_pipeline.run([c(), o()]))
+    print(duck_pipeline.dataset.df(table="customers"))
```

Comment:
This is super close. I'd modify this a little bit (following the ticket #1095)
access data in a table
`dataset.customers` or `dataset["customers"]` or `dataset.sql("SELECT * FROM customers")` creates a relation. relation is leazily evaluated. that happens when those methods are called:
* df()
* iter_df(batch_size) - iterator
* arrow()
* iter_arrow(batch_size)
* fetch() -> db api fetch all
* iter_fetch(batch_size) -> fetchmany
* ibis() - converts relation to IBIS relation via https://ibis-project.org/how-to/extending/sql

plugging in IBIS:
(not sure about that)
* dataset.df() - IBIS client on which we can get Table relation
* dataset.df("customer") - IBIS table relation on customers
* datasert.customers.ibis() - same as above

Reply:
I suggest these adjustments:

- `df()` -> `pandas()`
- `iter_df` -> `iter_pandas()`

Because:
- there are other dataframe flavors besides Pandas, e.g. Polars and Ibis
- it's consistent with `arrow()` and `ibis()` (they are also project names)

Reply:
makes sense to me, @rudolfix wdyt?